tests/integration/SdkTest_test.cpp

```diff
@@ -1265,6 +1293,67 @@ bool veryclose(double a, double b)
     return ratio * 1000000 < 1;
 }
 
+TEST_F(SdkTest, SdkTestKillSession)
+{
+    // Convenience.
+    using MegaAccountSessionPtr =
+      std::unique_ptr<MegaAccountSession>;
+
+    // Make sure environment variable are restored.
+    auto accounts = makeScopedValue(envVarAccount, string_vector(2, "MEGA_EMAIL"));
+    auto passwords = makeScopedValue(envVarPass, string_vector(2, "MEGA_PWD"));
+
+    // Get two sessions for the same account.
+    getAccountsForTest(2);
+
+    // Make sure the sessions aren't reused.
+    gSessionIDs[0] = "invalid";
+    gSessionIDs[1] = "invalid";
+
+    // Get our hands on the second client's session.
+    MegaHandle sessionHandle = UNDEF;
+
+    auto result = synchronousGetExtendedAccountDetails(1, true);
+    ASSERT_EQ(result, MegaError::API_OK)
+      << "GetExtendedAccountDetails failed (error: "
+      << result
+      << ")";
+
+    for (int i = 0; i < mApi[1].accountDetails->getNumSessions(); )
+    {
+        MegaAccountSessionPtr session;
+
+        session.reset(mApi[1].accountDetails->getSession(i++));
+
+        if (session->isCurrent())
+        {
+            sessionHandle = session->getHandle();
+            break;
+        }
+    }
+
+    // Were we able to retrieve the second client's session handle?
+    ASSERT_NE(sessionHandle, UNDEF)
+      << "Unable to get second client's session handle.";
+
+    // Kill the second client's session (via the first.)
+    result = synchronousKillSession(0, sessionHandle);
+    ASSERT_EQ(result, MegaError::API_OK)
+      << "Unable to kill second client's session (error: "
+      << result
+      << ")";
+
+    // Wait for the second client to become logged out.
+    ASSERT_TRUE(WaitFor([&]()
+                        {
+                            return mApi[1].megaApi->isLoggedIn()  == 0;
+                        },
+                        8 * 1000));
```

Comment:
Perhaps it would had been more consistent/robust to wait for the `API_ESID` received by the second client, which is notified to the app via a `onRequestFinish(TYPE_LOGOUT)`. In other words, closing the current session remotely, via another client, triggers a logout. See `void MegaApiImpl::request_error(error e)`